apiserver/common/networkingcommon/types.go

```diff
@@ -174,6 +174,9 @@ func NetworkInterfacesToStateArgs(devs corenetwork.InterfaceInfos) (
 			devicesArgs = append(devicesArgs, args)
 		}
 
+		if dev.PrimaryAddress().Value == "" {
```

Comment:
Is this just a sanity check or is it to filter devices that are down or have been `ip flush`'d?

Reply:
So here, we will always add the device (see the block above), but the way that the machine agent populates/transforms these means that we sometimes get a device without addresses, plus a copy of the device with each different address.

All this does is prevent the block below it from logging out warnings for ignoring the device (these are currently occurring without the change).